src/Component.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\twig_fractal\Component.
+ */
+
+namespace Drupal\twig_fractal;
+
+use Drupal\Component\Utility\Html;
+use Symfony\Component\Yaml\Yaml;
+
+class Component {
+
+  private static $instance;
+
+  /**
+   * The pathname of the component to render.
+   *
+   * @var string
+   */
+  protected $pathname;
+
+  /**
+   * The name of the component to render.
+   *
+   * @var string
+   */
+  protected $component;
+
+  /**
+   * The names of the variants to render.
+   *
+   * @var array
+   */
+  protected $variants = [];
+
+  private function __construct($compound) {
+    list($this->pathname, $this->component, $this->variants) = $this->extractComponentParts($compound);
+  }
+
+  public static function getInstance($compound) {
+    if (!isset(static::$instance)) {
+      static::$instance = new static($compound);
+    }
+    return static::$instance;
+  }
+
+  public function __get($name) {
```

Comment:
@sun, what would be a good practice for accessing class properties in this case?Having a magic method like